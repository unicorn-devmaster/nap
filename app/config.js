const config = {}

config.redis = {
  url: process.env.REDIS_STORE_URI,
  secret: process.env.EXPRESS_SESSION_SECRET
}

config.mongo = {
  url: process.env.MONGO_URI
}

module.exports = config