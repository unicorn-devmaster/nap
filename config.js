const config = {}

config.redis = {
  url: process.env.REDIS_STORE_URI,
  secret: process.env.EXPRESS_SESSION_SECRET
}

module.exports = config