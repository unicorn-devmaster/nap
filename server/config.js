// Helper
const bool = (str) => (str == void 0) ? false : str.toLowerCase() === 'true' // eslint-disable-line
const int = (str) => (!str) ? 0 : parseInt(str, 10) // eslint-disable-line
const float = (str) => (!str) ? 0: parseFloat(str, 10)  // eslint-disable-line

const dev = process.env.NODE_ENV !== 'production'

const config = {
  // Environments
  dev,

  // Passport
  mailgun_api_key: process.env.MAILGUN_API_KEY,
  mailgun_domain: process.env.MAILGUN_DOMAIN,

  // Services
  redis_url: process.env.REDIS_URI || 'redis://redis',
  mongo_url: process.env.MONGODB_URI || 'mongodb://mongo/graphql',
  port: int(process.env.PORT) || 3000,

  // Security
  cookie_secret: process.env.COOKIE_SECRET || 'foo',
  jwt_secret: process.env.JWT_SECRET || 'foo',
  
  passport_disabled: int(process.env.PASSPORT_DISABLED || '0') === 1,
  graphql_disabled: int(process.env.GRAPHQL_SERVER_DISABLED || '0') === 1,
  graphiql_enabled: process.env.GRAPHIQL_ENABLED !== undefined ? int(process.env.GRAPHIQL_ENABLED) === 1 : dev,

  mubsub: process.env.MUBSUB_URI,
  mubsub_enabled: process.env.MUBSUB_URI !== undefined && !!process.env.MUBSUB_URI,
}

module.exports = config
