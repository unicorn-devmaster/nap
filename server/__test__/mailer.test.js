/* eslint-env jest */

process.env.MAILGUN_API_KEY = 'FOO_MAILGUN_API_KEY'
process.env.MAILGUN_DOMAIN = 'BAR_MAILGUN_DOMAIN'

require('../config')

describe('mailer', () => {
  it('should return error if no API_KEY or DOMAIN provide while verification', async () => {
    const { willSendVerification } = require('../mailer')
    await willSendVerification(null, null).catch(err => {
      expect(err).toMatchSnapshot()
    })
  })

  it('should return error if no API_KEY or DOMAIN provide while reset password', async () => {
    const { willSendPasswordReset } = require('../mailer')
    await willSendPasswordReset(null, null).catch(err => {
      expect(err).toMatchSnapshot()
    })
  })

  it('should return verification mail body', async () => {
    const { createVerificationURL } = require('../passport-local')
    const email = 'katopz@gmail.com'
    const baseURL = 'http://localhost:3000'
    const token = 'FOO_TOKEN'
    const verificationURL = createVerificationURL(baseURL, token)

    const { willSendVerification } = require('../mailer')
    const payload = await willSendVerification(email, verificationURL)

    expect(payload).toMatchSnapshot()
  })

  it('should return reset mail body', async () => {
    const { createPasswordResetURL, createNewPasswordResetURL } = require('../passport-local')
    const email = 'katopz@gmail.com'
    const baseURL = 'http://localhost:3000'
    const token = 'FOO_TOKEN'
    const passwordResetURL = createPasswordResetURL(baseURL, token)
    const newPasswordResetURL = createNewPasswordResetURL(baseURL)

    const mailer = require('../mailer')
    const msg = await mailer.willSendPasswordReset(email, passwordResetURL, newPasswordResetURL)

    expect(msg).toMatchSnapshot()
  })
})