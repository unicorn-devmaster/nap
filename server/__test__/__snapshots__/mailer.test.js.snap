// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mailer should return error if no API_KEY or DOMAIN provide while reset password 1`] = `[Error: No password reset URL]`;

exports[`mailer should return error if no API_KEY or DOMAIN provide while verification 1`] = `[Error: No verification URL]`;

exports[`mailer should return reset mail body 1`] = `"FOO_MAILGUN_BODY"`;

exports[`mailer should return verification mail body 1`] = `"FOO_MAILGUN_BODY"`;
